src/spec-utils/httpRequest.ts

```diff
@@ -20,11 +20,17 @@ export function request(options: { type: string; url: string; headers: Record<st
 			headers: options.headers,
 			agent: new ProxyAgent(),
 		};
-		const req = https.request(reqOptions, res => {
+
+		const plainHTTP = parsed.protocol === 'http:' || parsed.hostname === 'localhost';
```

Comment:
Looks like this part of code to find request (http/https) is repetitive. Suggestion: Can we add a new function?

Reply:
I don't think this is too repetitive?  I do think that we may not need three functions here, but the first function throws and that behavior is expected across the codebase, which is why I didn't want to modify that behavior and had made a new function.

Reply:
'localhost' could be https for testing, would it make sense to drop that part of the condition?

Reply:
That's a good point. The main reason I added this in was that we have a lot of handcrafted URLs with `https://....` across the codebase, and at each of those points I'd need to determine if I wanted http or https.  To avoid a lot of redundant changes in this PR, I just look for `localhost`. 

I'll make a note to clean this up but I think it's fine for now?